src/controllers/userController.ts

```diff
@@ -0,0 +1,46 @@
+import { Request, Response } from 'express';
+import { User } from '../models/user';
+import { UserRepository } from '../repository/userRepository';
+import { BaseController } from './baseController';
+import { Validations } from '../validations';
+
+const checkRequest = new Validations;
+
+export class UserController extends BaseController {
+
+    userRepository: UserRepository;
+
+    constructor(userRepository: UserRepository) {
+        super();
+        this.userRepository = userRepository;
+    }
+    
+    signUp(req: Request, res: Response) {
+        const { email, password } = req.body;
+
+        checkRequest.validateEmailPassword(email, password)
+        .then((errors) => {
+            if (errors.length === 0) {
+                return this.userRepository.findUser(email)
+                .then((user) => {
+                    if (user === undefined) {
+                        const user = new User({ email: email, password: password });
+                        return this.userRepository.save(user)
+                        .then(() => res.json({ message: "user created with success" }))
+                        .catch((error) => super.handleErrors(res, error));
```

Comment:
this `catch` should be outside, at the end of the promise chain, so you are catching any possible issue that could happen besides the `this.userRepository.save(user)`